### DISCUSSION TITLE
Prombles about on_click

### DISCUSSION THREAD

**Discussion by umarurize** [EndstoneMC/endstone#84] [Q&A] [answered]
Code preview
![059827ec5d7f5ff86214f54d09cbf87](https://github.com/user-attachments/assets/a2681d6d-0cce-449d-aee0-43e1a7925aa2)
I hope my code can execute like what the picture show, but it seems that the function (confirm(), on_transfer()) behind the on_click will be automatically executed. I originally hope these functions can be executed after the player clicks.

I have tried many ways to handel this problem but failed finally, hoping author can help me deal with this. @wu-vincent

**Comment by umarurize** [comment, marked as answer]
Finally, i found the easiest way to deal with this problem, just add lambda _: after on_click...

**Reply by xXLarryTFVWXx** [reply]
Because it on_click needs to be a "callable". The old way you were doing it was giving on_click the result of the confirm function, not a "callable" object.

**Reply by umarurize** [reply]
yes, you are right,  I later carefully read the document and understood this point. Anyway, thank you for answering my question!

**Comment by ye111566** [comment]
你可以用添加类内变量的方法来完成传入除了player和json_str以外的参数
